Makefile

```diff
@@ -165,16 +165,28 @@ ${BIN_DIR}/deepcopy-gen:
 	$(info Compiling deepcopy-gen...)
 	go build -o ${BIN_DIR}/deepcopy-gen ./cmd/deepcopy-gen/
 
+
+# Build the docker image and tag it.  
 images: verify-tag-name generate-code update-deployment-crds
 	$(info List executable directory)
 	$(info repo id: ${git_repository_id})
 	$(info branch: ${GIT_BRANCH})
 	$(info Build the docker image)
-ifeq ($(strip $(GO_BUILD_ARGS)),)
-	docker build --quiet --no-cache --tag mcad-controller:${TAG} -f ${CURRENT_DIR}/Dockerfile  ${CURRENT_DIR}
-else 
-	docker build --no-cache --tag mcad-controller:${TAG} --build-arg GO_BUILD_ARGS=$(GO_BUILD_ARGS) -f ${CURRENT_DIR}/Dockerfile  ${CURRENT_DIR}
-endif		
+	echo "Host architecture: $(shell uname -m)"
+	@HOST_ARCH=$(shell uname -m); \
+	if [ "$$HOST_ARCH" = "arm64" ]; then \
+		if [ "$(strip $(GO_BUILD_ARGS))" = "" ]; then \
```

Comment:
Could this get moved to a separate PR? It's not strictly related to this PR, and we would be able to iterate on that part without blocking the work on the metrics part.

Reply:
you're right here, to alter this is the only way I could test my changes, hence in this PR but I can move it now 👍